### PR TITLE
feat: Add time-series history to HomeBlaze

### PR DESCRIPTION
**Pull request by RicoSuter**
Adds time-series history to HomeBlaze: eligible `[State]` properties are recorded by independent history stores, queried through a stateless cross-store merger, and surfaced in a property history dialog and a `get_property_history` MCP tool. Two stores ship, In-Memory (priority 100, full-resolution live edge) and SQLite (priority 50, durable local history); a TimescaleDB tier is designed but not built.

Two supporting changes ride along because history depends on them. Every user-facing timestamp now renders in a viewer-selected display timezone rather than the server's, which is what makes a history chart readable at all on a remote server. And the Philips Hue integration was reworked so its derived state is observable through interception, since state that never raises a change cannot be recorded.

It also fixes the E2E fixture, which opened a browser context per test and closed none of them until the suite ended. Each open page holds a live Blazor circuit that the host re-renders on every state change, and the resulting pile of idle circuits was making the last E2E test miss its clicks on CI.

Supersedes #317 and #249, which were the two planning branches for this work.

## Why

HomeBlaze could show the current value of a property but nothing about how it got there. Charts, historical queries, and AI questions about the past all need recorded samples, and a home installation and an industrial one want different storage.

The design avoids a central history coordinator. Each store is an `[InterceptorSubject]` `BackgroundService` that records and serves its own data and publishes its own coverage, matching how connectors are already configured, and a stateless merger stitches results by priority and coverage. The store engines are graph-free and operate on canonical paths and typed values, so storage logic stays testable without a HomeBlaze graph and can be extracted into a generic library later.

Recording correctness rests on coverage rather than on the presence of samples. A coverage range asserts that the store was collecting its configured stream throughout that interval and detected no loss, so absence of a sample means the property did not change. That distinction is what lets the chart draw "no data" differently from "no value", and it is why restarts, bounded-queue overflows, and retention all have explicit effects on coverage.

## Contract

For anyone implementing an `IHistoryStore`:

- `CoverageRanges` is an immutable, ordered, non-overlapping snapshot of half-open `[From, To)` intervals. A range asserts the store was collecting throughout it and detected no loss; it is store-wide and all-or-nothing, never per property.
- A store must install its change subscription before beginning a coverage session, so no change can fall inside claimed coverage without reaching the engine.
- A store that drops a sample must end its current range at the first dropped change and start a new one only after it is healthy again. Coverage must never extend past data that is not durable.
- Queries require a non-empty path and aggregation, `From < To`, a positive bucket when given, and a positive `MaxPoints`. Stores return the newest `MaxPoints` points in ascending timestamp order and set `Truncated` when older results were omitted.
- Query errors propagate. A failed store must not be reported as an empty store.

For callers of the merger:

- A bucket is served by exactly one store, never split, so aggregates are never combined across stores.
- Uncovered buckets are explicit null points rather than omitted ones, and carried state (`Last`, `TimeWeightedAverage`) resets across an uncovered interval instead of interpolating through it.
- Numeric aggregations on JSON-backed properties throw `HistoryAggregationNotSupportedException`.

## Breaking changes

All of these are confined to `src/HomeBlaze`, which is not packaged. They matter to plugin authors compiling against these assemblies, not to NuGet consumers of the Namotion.Interceptor libraries, which this branch does not touch.

- API: `SubjectPathResolverExtensions.WithPathResolver()` is removed, and `SubjectPathResolver` now takes `Func<IInterceptorSubject?>` instead of `RootManager`. `RootManager` needs the resolver registered in the context before it loads the graph, so taking `RootManager` in the constructor was a cycle. The resolver is registered by the standard HomeBlaze service registration; callers that invoked `WithPathResolver()` should drop the call.
- API: `McpBuilderExtensions.WithHomeBlazeMcpTools` gained a trailing `params Func<McpToolProviderContext, IMcpToolProvider>[]`, which is how the history package contributes its tool without `HomeBlaze.AI` referencing the history stack. Source compatible; a precompiled caller must be recompiled.
- Behavior: Hue operations now throw on an API error instead of returning silently, and no longer write the expected value into the local resource before the bridge confirms it. State arrives on the next poll. A caller that relied on the operation being silent on failure, or on the value being readable immediately after the call returns, has to change.
- Behavior: the shipped sample Hue bridge configuration polls every 60 seconds instead of every 500 ms. An existing `Hue.json` is not modified; only the shipped default changed.

`StateUnit.Byte` was added to the enum, which is additive.

The five new `VerifyChecksTests.PublicApi.verified.txt` snapshots are all new files for the new history packages. No existing snapshot changed.

## Diff composition

| Area | Files | Added | Removed | Net |
|---|---:|---:|---:|---:|
| Production code | 70 | 8'554 | 134 | +8'420 |
| Tests and benchmarks | 74 | 9'102 | 46 | +9'056 |
| Documentation | 6 | 440 | 78 | +362 |
| Project and CI files | 12 | 193 | 1 | +192 |
| **Total** | **162** | **18'289** | **259** | **+18'030** |

## Verification

- [x] Documentation updated
- [x] Unit tests
- [x] Integration tests
- [ ] ~~Benchmarks~~ nothing in the benchmark project reaches this change; no core library code is touched. Outside `src/HomeBlaze` the branch only adds the new projects to `Namotion.Interceptor.slnx` and a `.codex/config.toml` pointing Codex at the local HomeBlaze MCP endpoint through an exported dev certificate instead of disabling TLS verification
- [ ] ~~Load tests~~ the Connector Tester exercises the connector stack, which this change does not modify
- [ ] ~~Chaos tests~~ same reason

Documentation: `history.md` rewritten for the implemented v1 design, `localization.md` added for the display timezone, and `ai.md`, `overview.md`, `state.md` and `versioning.md` updated where they referenced the changed behavior.

Unit tests: `dotnet test src/Namotion.Interceptor.slnx --filter "Category!=Integration"`, 3'770 passed, 0 failed, across 34 test assemblies.

Integration tests: `dotnet test src/HomeBlaze/HomeBlaze.E2E.Tests`, 26 passed, 0 failed, including the Playwright coverage of the property history dialog. `Validation_NameErrors_PreventAdvancing` had been failing on roughly a third of CI runs while never reproducing locally; that is the fixture leak described above, and it is fixed here rather than left for a follow-up.

Not run: the Connector Tester load and chaos profiles, for the reason above.

### CODE REVIEW

File: src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs [hunk 1]
```diff
@@ -0,0 +1,1198 @@
...
+namespace HomeBlaze.History.Sqlite;
+
+/// <summary>
+/// The graph-free, SQL-backed history engine. Operates only on canonical path strings and typed
+/// values: partition-file management, schema, batched write plus periodic flush, raw queries,
+/// look-back, coverage, and metrics. Mirrors the value routing and point mapping of
+/// <c>InMemoryHistoryStoreCore</c> so query results are identical, but persists rows into
+/// partitioned SQLite database files with <c>value_json</c> stored as TEXT.
+/// </summary>
+internal sealed class SqliteHistoryStoreCore : IDisposable
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs]: shouldnt we use async apis for sqlite queries etc.?

File: src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs [hunk 2]
```diff
@@ -0,0 +1,1198 @@
...
+{
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs]: can we split this into multiple classes (eg 300 locs each)? is there a good seam for that?

File: src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs (hunk 1 above)
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs]: shouldnt this also implement IHistoryStore so that we can later move it to Namotion.Interceptor.History.*? Not sure the postfix *Core is good, maybe the subject should have the *Subject postfix and core removed? or better idea?

File: src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryAggregations.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    public const string Maximum = "Maximum";
+
+    /// <summary>Sum of values in the bucket.</summary>
+    public const string Sum = "Sum";
+
+    /// <summary>Number of samples in the bucket.</summary>
+    public const string Count = "Count";
+
+    /// <summary>Sample standard deviation.</summary>
+    public const string StdDev = "StdDev";
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryAggregations.cs]: maybe we should not use abbreviations?

File: src/HomeBlaze/HomeBlaze.History/HistoryStoreMerger.cs [hunk 4]
```diff
@@ -0,0 +1,507 @@
...
+namespace HomeBlaze.History.Abstractions;
+
+/// <summary>
+/// Stateless cross-store query merger. Resolves a single <see cref="HistoryQuery"/> against a set
+/// of <see cref="IHistoryStore"/> instances by planning non-overlapping sub-queries (raw coverage
+/// subtraction or per-bucket single-owner dispatch) and executing them under a shared point budget.
+/// The merger is a pure function over the store set, which is the extraction-ready seam: the host
+/// decides where the set comes from (HomeBlaze: the registry's known subjects).
+/// </summary>
+public static class HistoryStoreMerger
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.History/HistoryStoreMerger.cs:14]: not sure this belongs to history.abstractions, maybe should be in homeblaze.history? abstractions should only be about interfaces, or is this needed for some ref reasons or because it's good design?

File: src/HomeBlaze/HomeBlaze.History.Abstractions/IHistoryStore.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    /// </summary>
+    int Priority { get; }
+
+    /// <summary>
+    /// Gets the time window this store can answer queries for.
+    /// </summary>
+    HistoryCoverage Coverage { get; }
+
+    /// <summary>
+    /// Gets the aggregation identifiers (see <see cref="HistoryAggregations"/>) this store supports.
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.History.Abstractions/IHistoryStore.cs:29]: maybe we change it to ... current... so it's clear that it is a snapshot and might change for every read? or is that clear?

File: src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryAggregations.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    /// <summary>
+    /// Aggregations every store supports regardless of column type. The global eligibility
+    /// check skips these (see the cross-store merger).
+    /// </summary>
+    public static readonly IReadOnlySet<string> Universal =
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryAggregations.cs]: Find better name for Universal, isnt clear from naming i think.

File: src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryColumns.cs [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+/// <summary>
+/// Single source of truth for routing a value into a column (write) and building
+/// column-targeted SQL (read).
+/// </summary>
+public static class HistoryColumns
+{
+    /// <summary>
+    /// Returns the column a value of <paramref name="propertyType"/> is stored in.
+    /// </summary>
+    public static ValueColumn ValueColumnFor(Type propertyType)
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryColumns.cs]:
```suggestion
    public static ValueColumn GetValueColumnFor(Type propertyType)
```

File: src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryStoreMerger.cs [hunk 8]
```diff
@@ -0,0 +1,507 @@
...
+    /// For bucketed queries the check runs against the bucket-aligned span the planner actually
+    /// enumerates (from the start of the bucket containing <c>From</c> through the end of the bucket
+    /// containing <c>To</c>), not the raw <c>[From, To)</c>. The bucketed planner assigns each
+    /// epoch-aligned bucket only to a store whose coverage fully contains that bucket, so when
+    /// <c>From</c>/<c>To</c> are not bucket-aligned the edge buckets extend outside <c>[From, To)</c>.
+    /// Checking the aligned span keeps eligibility and per-bucket ownership in agreement: eligibility
+    /// throws iff some bucket the planner would enumerate has no supporting store that fully contains
+    /// it (otherwise that bucket would be a silent gap the eligibility check had promised away).
+    /// </summary>
+    internal static void CheckEligibility(IReadOnlyList<IHistoryStore> ordered, HistoryQuery query)
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryStoreMerger.cs]:
```suggestion
    internal static void EnsureEligibility(IReadOnlyList<IHistoryStore> ordered, HistoryQuery query)
```

File: src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs (hunk 1 above)
Comment (RicoSuter) [reply src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs]: Decided to keep SQLite synchronous. Microsoft.Data.Sqlite's *Async methods run synchronously (SQLite is a synchronous in-process library), and the store serializes all connection access under a single re-entrant lock, so going async would mean reworking that lock into a SemaphoreSlim for no real I/O gain. The async IHistoryStore contract is satisfied via Task.FromResult. The upcoming TimescaleDB store will be genuinely async via Npgsql.

File: src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs (hunk 2 above)
Comment (RicoSuter) [reply src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs]: Done in 57faf340: split into SqliteHistoryStore (engine, lock + connections + orchestration), SqliteValueRouting, SqliteHistoryWriter, SqliteHistoryReader (raw + look-back + chain), and SqliteBucketReader (bucketed + TWA).

File: src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs (hunk 1 above)
Comment (RicoSuter) [reply src/HomeBlaze/HomeBlaze.History.Sqlite/SqliteHistoryStoreCore.cs]: Done in 17ca6f1e: the graph-free engine is now the public SqliteHistoryStore : IHistoryStore (the extraction unit ready to lift into a generic library), and the [InterceptorSubject] adapter is renamed SqliteHistoryStoreSubject, which still implements IHistoryStore by delegating so KnownSubjects discovery is unchanged. Same restructure for the InMemory store.

File: src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryAggregations.cs (hunk 3 above)
Comment (RicoSuter) [reply src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryAggregations.cs]: Done in b369b5a4: StdDev renamed to StandardDeviation (constant and value).

File: src/HomeBlaze/HomeBlaze.History/HistoryStoreMerger.cs (hunk 4 above)
Comment (RicoSuter) [reply src/HomeBlaze/HomeBlaze.History/HistoryStoreMerger.cs:14]: Done in 4c34ffc8: HistoryStoreMerger moved to a new HomeBlaze.History package. HomeBlaze.History.Abstractions now keeps only the interface, the DTOs, and the pure helpers (HistoryColumns, BucketAlignment, HistoryEligibility, HistoryAggregations).

File: src/HomeBlaze/HomeBlaze.History.Abstractions/IHistoryStore.cs (hunk 5 above)
Comment (RicoSuter) [reply src/HomeBlaze/HomeBlaze.History.Abstractions/IHistoryStore.cs:29]: Done in b369b5a4: IHistoryStore.Coverage renamed to CurrentCoverage, with the doc comment clarifying it is a per-read snapshot that can change between reads.

File: src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryAggregations.cs (hunk 6 above)
Comment (RicoSuter) [reply src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryAggregations.cs]: Done in b369b5a4: Universal renamed to AlwaysAvailable.

File: src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryColumns.cs (hunk 7 above)
Comment (RicoSuter) [reply src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryColumns.cs]: Done in b369b5a4: applied (ValueColumnFor renamed to GetValueColumnFor).

File: src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryStoreMerger.cs (hunk 8 above)
Comment (RicoSuter) [reply src/HomeBlaze/HomeBlaze.History.Abstractions/HistoryStoreMerger.cs]: Done in b369b5a4: applied (CheckEligibility renamed to EnsureEligibility).

File: src/HomeBlaze/HomeBlaze.AI/Mcp/HistoryToolParsing.cs [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+namespace HomeBlaze.AI.Mcp;
+
+/// <summary>
+/// Pure parsing and normalization for the <c>get_property_history</c> MCP tool. Kept separate so
+/// the boundary parsing is unit-testable without a live MCP server or registry.
+/// </summary>
+public static class HistoryToolParsing
+{
+    /// <summary>All recognized aggregation identifiers (canonical PascalCase).</summary>
+    public static readonly IReadOnlySet<string> AllAggregations = new HashSet<string>(StringComparer.Ordinal)
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.AI/Mcp/HistoryToolParsing.cs]: Would it be better to dynamically resolve these from the available stores (union)?

File: docs/superpowers/specs/2026-06-07-history-design.md [hunk 10]
```diff
@@ -0,0 +1,573 @@
+# Time-Series History Design (Planning Spec)
```
Comment (RicoSuter) [inline docs/superpowers/specs/2026-06-07-history-design.md]: delete this file

File: src/HomeBlaze/Directory.Build.props [hunk 11]
```diff
@@ -1,4 +1,8 @@
 <Project>
+  <!-- MSBuild stops at the nearest Directory.Build.props, so without this import the HomeBlaze tree
```
Comment (RicoSuter) [inline src/HomeBlaze/Directory.Build.props]: remove comment, its clear

File: src/HomeBlaze/HomeBlaze/Data/Docs/architecture/design/ai.md [hunk 12]
```diff
@@ -34,7 +34,7 @@ MCP tools are split across two packages, configured via a single `McpServerConfi
 | Package | Tools | Scope |
 |---------|-------|-------|
 | `Namotion.Interceptor.Mcp` | `query`, `get_property`, `set_property`, `list_types` | Any Namotion.Interceptor application |
-| `HomeBlaze.AI` | `list_methods`, `invoke_method` (via `IMcpToolProvider`) + subject enrichment (`$type`, `$icon`, `$title` via `IMcpSubjectEnricher`) + concrete type discovery (via `IMcpTypeProvider`). History tools (`get_property_history`, `get_event_history`, `get_command_history`) planned for later | HomeBlaze-specific features |
+| `HomeBlaze.AI` | `list_methods`, `invoke_method` (via `IMcpToolProvider`) + subject enrichment (`$type`, `$icon`, `$title` via `IMcpSubjectEnricher`) + concrete type discovery (via `IMcpTypeProvider`) + `get_property_history`. Event and command history tools remain planned | HomeBlaze-specific features |
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze/Data/Docs/architecture/design/ai.md]: keep method/tool names in docs (events and commands)?

File: src/HomeBlaze/HomeBlaze.Services/ServiceCollectionExtensions.cs [hunk 13]
```diff
@@ -26,7 +26,17 @@ public static IServiceCollection AddHomeBlazeServices(this IServiceCollection se
...
-        services.AddHostedService(sp => sp.GetRequiredService<RootManager>());
+        services.AddScoped<ITimeZoneDisplay, TimeZoneDisplayService>();
+        services.AddHostedService(sp =>
+        {
+            // SubjectPathResolver registers itself as a context service in its constructor, and
+            // subjects attached by RootManager (history stores) need it the moment their hosted
+            // service starts. Force the singleton before RootManager loads the graph; otherwise the
+            // resolver would first be constructed by whichever Blazor component injects it, which
+            // happens after startup.
+            sp.GetRequiredService<SubjectPathResolver>();
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.Services/ServiceCollectionExtensions.cs]: is there a cleaner approach here? looks a bit hacky.

File: src/HomeBlaze/HomeBlaze.Host.Services/Display/StateUnitExtensions.cs [hunk 14]
```diff
@@ -17,7 +18,7 @@ private static readonly (StateUnit Unit, string Suffix, decimal Factor)[][] Unit
         [(StateUnit.WattHour, "Wh", 1m), (StateUnit.KilowattHour, "kWh", 1000m)],
         [(StateUnit.Millimeter, "mm", 1m), (StateUnit.Meter, "m", 1000m), (StateUnit.Kilometer, "km", 1_000_000m)],
         [(StateUnit.Milliampere, "mA", 1m), (StateUnit.Ampere, "A", 1000m)],
-        [(StateUnit.Kilobyte, "kB", 1m), (default, "MB", 1000m), (default, "GB", 1_000_000m)],
+        [(StateUnit.Byte, "B", 1m), (StateUnit.Kilobyte, "kB", 1000m), (default, "MB", 1_000_000m), (default, "GB", 1_000_000_000m)],
```
Comment (RicoSuter) [inline src/HomeBlaze/HomeBlaze.Host.Services/Display/StateUnitExtensions.cs:21]: should this be 1024, 1024x1024, etc?

File: src/HomeBlaze/Namotion.Devices.Philips.Hue/HueResourceEcho.cs [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+/// Applies the optimistic local echo after a successful bridge command.
+///
+/// The Hue SDK resources are plain mutable DTOs held behind an intercepted property. Mutating one in
+/// place (<c>LightResource.On.IsOn = true</c>) changes no intercepted property, so nothing recomputed
+/// the derived state, nothing notified the UI and nothing recorded a history sample: the change only
+/// appeared after the next bridge poll, up to a minute later. Re-assigning the same instance does not
+/// help either, because the equality check vetoes a write of the value already there. So the change is
+/// applied to a copy and the copy is assigned, which is exactly what the poll path does.
+/// </summary>
+internal static class HueResourceEcho
```
Comment (RicoSuter) [inline src/HomeBlaze/Namotion.Devices.Philips.Hue/HueResourceEcho.cs]: echo seems of, maybe just HueResource or HueResourceMutation or something like that?
